exasol_transformers_extension/utils/bucketfs_operations.py

```diff
@@ -1,62 +1,93 @@
-import subprocess
+from __future__ import annotations
 import tarfile
 import tempfile
 from pathlib import PurePosixPath, Path
 from typing import BinaryIO
 
-from exasol_bucketfs_utils_python.abstract_bucketfs_location import \
-    AbstractBucketFSLocation
-from exasol_bucketfs_utils_python.bucket_config import BucketConfig
-from exasol_bucketfs_utils_python.bucketfs_config import BucketFSConfig
-from exasol_bucketfs_utils_python.bucketfs_connection_config import \
-    BucketFSConnectionConfig
-from exasol_bucketfs_utils_python.bucketfs_factory import BucketFSFactory
-from exasol_bucketfs_utils_python.bucketfs_location import BucketFSLocation
-from tenacity import retry, wait_fixed, stop_after_attempt
+import json
+import exasol.bucketfs as bfs
+from exasol.saas.client.api_access import get_database_id   # type: ignore
 
 
-def create_bucketfs_location_from_conn_object(bfs_conn_obj) -> BucketFSLocation:
-    return BucketFSFactory().create_bucketfs_location(
-        url=bfs_conn_obj.address,
-        user=bfs_conn_obj.user,
-        pwd=bfs_conn_obj.password)
+def create_bucketfs_location_from_conn_object(bfs_conn_obj) -> bfs.path.PathLike:
+
+    bfs_params = json.loads(bfs_conn_obj.address)
+    bfs_params.update(json.loads(bfs_conn_obj.user))
+    bfs_params.update(json.loads(bfs_conn_obj.password))
+    return bfs.path.build_path(**bfs_params)
 
 
 def create_bucketfs_location(
-        bucketfs_name: str, bucketfs_host: str, bucketfs_port: int,
-        bucketfs_use_https: bool, bucketfs_user: str, bucketfs_password: str,
-        bucket: str, path_in_bucket: str) -> BucketFSLocation:
-    _bucketfs_connection = BucketFSConnectionConfig(
-        host=bucketfs_host, port=bucketfs_port, user=bucketfs_user,
-        pwd=bucketfs_password, is_https=bucketfs_use_https)
-    _bucketfs_config = BucketFSConfig(
-        bucketfs_name=bucketfs_name, connection_config=_bucketfs_connection)
-    _bucket_config = BucketConfig(
-        bucket_name=bucket, bucketfs_config=_bucketfs_config)
-    return BucketFSLocation(
-        bucket_config=_bucket_config,
-        base_path=PurePosixPath(path_in_bucket))
+        path_in_bucket: str = '',
+        bucketfs_name: str | None = None,
+        bucketfs_url: str | None = None,
+        bucketfs_host: str | None = None,
+        bucketfs_port: int | None = None,
+        bucketfs_use_https: bool = True,
+        bucketfs_user: str | None = None,
+        bucketfs_password: str | None = None,
+        bucket: str | None = None,
+        saas_url: str | None = None,
+        saas_account_id: str | None = None,
+        saas_database_id: str | None = None,
+        saas_database_name: str | None = None,
+        saas_token: str | None = None,
+        use_ssl_cert_validation: bool = False
+) -> bfs.path.PathLike:
+
+    # Infer where the database is - on-prem or SaaS.
+    if all((any((bucketfs_url, all((bucketfs_host, bucketfs_port)))), bucketfs_name,
```

Comment:
If you would assign the expression to a variable with this would be way easier to read, e.g.:

```python
is_on_prem = all(...)
if is_on_prem:
   pass
else:
   pass
```